### PR TITLE
Proposed Abstract classes and methods

### PR DESCRIPTION
**Pull request by dscreve**

### CODE REVIEW

**Comment by dscreve** [comment]
Proposed : abstract classes and methods

File: proposals/0000-abstract classes and methods.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+This change has no impact on existing code, but might change the ABI that is being 
+stabilizing in Swift 3.0.
+
+## Alternatives considered
+As first reading, it seems that protocols and protocol extensions might fit the need. It 
+actually does not because abstract classes can have attributs and properties that 
+protocols does not support.
+
+An alternative solution would be to add attributes to protocols and protocol extensions, 
+but this might broke compatibility with Objective-C runtime.
```
Comment (devashish2203) [inline proposals/0000-abstract classes and methods.md]: this might ~~broke~~ break compatibility

**Comment by DougGregor** [comment]
Has this been discussed on the swift-evolution mailing list yet?

**Comment by pedrovieira7** [comment]
Has this been submitted to the swift-evolution mailing list? I don't recall seeing this

**Comment by DougGregor** [comment]
I just checked, and it has not (at least not under an easily-findable subject). Closing the PR; please open up a discussion on swift-evolution

**Comment by dscreve** [comment]
Yes, we did and someone suggested to use protocol extensions.

David Scrève

**Comment by DougGregor** [comment]
Please send this as a [Proposal draft] to swift-evolution, since several of us didn't see and could not find the discussion you're talking about. It appears too early to schedule a public review.

**Comment by dscreve** [comment]
Here it is : https://lists.swift.org/pipermail/swift-evolution/2015-December/000123.html

But I think it was lost in the massive flow of early days.

I'll post a new proposal as you suggested.

**Pull request closed without merging**